packages/elements/src/react/common/form/otp.tsx

```diff
@@ -2,18 +2,17 @@ import { Control as RadixControl } from '@radix-ui/react-form';
 import * as React from 'react';
 
 import type { FormControlProps } from '~/react/common/form';
-import { useFocus } from '~/react/hooks';
 
 export type OTPInputProps = Exclude<
   FormControlProps,
   'type' | 'autoComplete' | 'minLength' | 'maxLength' | 'inputMode' | 'pattern' | 'spellCheck'
 > & {
-  render?: (props: { value: string; status: 'cursor' | 'selected' | 'none'; index: number }) => React.ReactNode;
+  render?: (props: { value: string; status: OTPInputSegmentStatus; index: number }) => React.ReactNode;
   length?: number;
   autoSubmit?: boolean;
 };
 
-type SelectionRange = readonly [start: number | null, end: number | null];
+type SelectionRange = readonly [start: number, end: number];
```

Comment:
I made the type closer to the actual implementation, since according to https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-textarea/input-selectionstart it can be `null`, too